### ISSUE TITLE
Problems with long filenames

### ISSUE THREAD

**Issue by mblataric** [jfromaniello/nestin#4]
Hi, 

When you try nest files with long filenames, combobox shows clipped text and it's not helpful at all. 
See attached image for example. 

Thanks, 
Mario

![nestinproblem](https://cloud.githubusercontent.com/assets/10866736/6061402/c1f80c04-ad47-11e4-8514-6d319d121893.jpg)

**Comment by jeremysimmons** [comment]
Hi Mario.
There are some options to fix this.
1. Rearrange the controls on the dialog so your long filename will fit inside the client rectangle of drop drop box.
2. truncate the filename by replacing characters the middle of the string with an elipsis (...)

Do you have a preference?

**Comment by mblataric** [comment]
Hi,

I think first option is better (fit file names into controls and optionaly make selected filename readonly editbox, which would allow cursor moving and scrolling through the filename).
Second option could be problematic if file difference would be in the middle.

Thanks,
Mario

**Comment by jeremysimmons** [comment]
Good catch on option 2 having the difference be in the middle.

![nestin](https://cloud.githubusercontent.com/assets/1406886/6064692/fcf02362-ad1c-11e4-86c7-84332cec00a9.png)
- moved label to title bar
- expand the drop down box and window size
- add a read-only edit box to display selected item, in case the text is larger than the combobox

**Comment by mblataric** [comment]
Hi,

Looks good. Try and see if you can make combobox editable (but readonly), that way you could ditch additional editbox. If not, this was is just fine.
Btw, what tool are you using for wireframes?

**Comment by jeremysimmons** [comment]
I don't think an combobox that allows editing is the right choice from a UX perspective.
The tool for mockups is www.balsamiq.com

**Comment by mblataric** [comment]
Editable combobox is rather standard UX element used in situations when you can either select something from list or enter it manually. 
I understand this is not situation in this case, but it would solve long filenames issue without adding an extra control. Of course "edit" control should be read-only, not editable. It should only allow accepting text cursor and movement through the text.

But, I'll be happy with two controls as well.

**Comment by markadrake** [comment]
Could a possible solution involve doing a string compare to filter the similar portions of the files and simple showing the difference?

Is there a way to add a context menu item so if you select a nested file you have the option to make it the root node?

All I can think of right now.

**Comment by mloeffen** [comment]
Wouldn't it suffice to set the FormBorderStyle to SizableToolWindow instead of the current setting (FixedDialog)?

**Comment by Myzifer** [comment]
Yep I found this being an issue too and couldn't be solved with renaming the file as they were for code first migrations which I was trying to nest based on the file types, if the window could be resized or the full text could be seen by hovering over then either would do.

Unfortunately I tried to do some trial and error to get the right file nested and then found out it refused to unnest the files.

I tried to unnest the top file, all lower files singularly and highlighting all files to right click and unnest but no right click options are available on the last option.

Next thing I tried was to uninstall/reinstall which half worked once but not again so I then found that using the extension "File Nesting" allowed me to fix it and does show full file names so my suggestion for anybody else with similar issues would be to do the same.
